web/views.py

```
@@ -305,6 +310,160 @@
+    model = Survey
+    template_name = "surveys/list.html"
+    login_url = "/login/"  # Set your login URL
+
+
+class SurveyCreateView(LoginRequiredMixin, CreateView):
+    model = Survey
+    form_class = SurveyForm
+    template_name = "surveys/create.html"
+    login_url = "/login/"
+
+    def form_valid(self, form):
```

Ruff (0.8.2)

`form_valid` is too complex (13 > 10)

(C901)